tests/socket.rs

```diff
@@ -531,6 +531,27 @@ fn out_of_band() {
     assert_eq!(unsafe { assume_init(&buf[..n]) }, DATA);
 }
 
+#[test]
+fn peek_sender_udp() {
+    let socket_a = Socket::new(Domain::IPV4, Type::DGRAM, Some(Protocol::UDP)).unwrap();
+    let socket_b = Socket::new(Domain::IPV4, Type::DGRAM, Some(Protocol::UDP)).unwrap();
+
+    let localhost_port_0 = SockAddr::from(SocketAddrV4::new(Ipv4Addr::LOCALHOST, 0));
+
+    // Allocate local address/port pairs for both sockets.
+    socket_a.bind(&localhost_port_0).unwrap();
+    socket_b.bind(&localhost_port_0).unwrap();
+
+    let socket_a_addr = socket_a.local_addr().unwrap();
+    let socket_b_addr = socket_b.local_addr().unwrap();
+
+    socket_b.send_to(b"Hello, world!", &socket_a_addr).unwrap();
+
+    let sender_addr = socket_a.peek_sender().unwrap();
+
+    assert_eq!(sender_addr.as_socket_ipv4(), socket_b_addr.as_socket_ipv4());
```

Comment:
This is failing on macOS because `sender_addr` is somehow _not_ IPv4. 

Unless the answer here is obvious to everyone else but me, I'm gonna get set up on an iMac Pro at work tomorrow to hit this with the debugger and figure out what's actually being returned here. I'd just edit the test to print it out but it seems inconsiderate to use CI minutes for debugging.

Reply:
It seems it doesn't work on macOS. We get an empty `SockAddr`:

```
SockAddr {
    ss_len: 0,
    ss_family: 0,
    len: 0,
}
```

Reply:
It may not like the zero-sized read buffer. I figured I'd pass something small but non-zero sized like 8 bytes.

Reply:
Fix it by passing a non-zero sized buffer, at least on macOS on ARM.